config/v1/types_insights.go

```diff
@@ -0,0 +1,72 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
```

Comment:
What would be the plural name of `Insights`?

Reply:
Good question. Any other name suggestion please?

Reply:
I'm keen to have a conversation about the names of these and the purpose of these objects. It seems to me currently that this object is a singleton which then requests multiple gather actions through its lifetime. That's not a usual pattern in Kubernetes, if this were instead a `Gather` which lifecycled with a single gather event (one collection of the data), then this would make it fit better into the Kube model. Has this idea been discussed (on an enhancement?) at all? And if so, may I ask why it was decided this is a better route

Reply:
Yes, you're right (it's a singleton possibly requesting multiple gather actions). AFAIK this idea hasn't been discussed in the corresponding enhancement.